qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -26,6 +26,147 @@
+
+        prev_outcome = "0" * num_qubits
+        state_strings = self._get_state_strings(num_qubits)
+
+        # Setup the list of dicts where each dict corresponds to a circuit.
+        sorted_memory = [{"memory": [], "metadata": circ.metadata} for circ in run_input]
+
+        for _ in range(shots):
+            for circ_idx, _ in enumerate(run_input):
+                prob = self._precomputed_probabilities[(circ_idx, prev_outcome)]
+                # Generate the next shot dependent on the pre-computed probabilities.
+                outcome = self._rng.choice(state_strings[:2], p=[1 - prob, prob])
+                # Append the single shot to the memory of the corresponding circuit.
+                sorted_memory[circ_idx]["memory"].append(hex(int(outcome, 2)))
+
+                prev_outcome = outcome
+
+        for idx, circ in enumerate(run_input):
+            ones = sorted_memory[idx]["memory"].count("0x1")
+            run_result = {
+                "shots": shots,
+                "success": True,
+                "header": {"metadata": circ.metadata},
+                "meas_level": meas_level,
+                "data": {
+                    "counts": {"1": ones, "0": shots - ones},
+                    "memory": sorted_memory[idx]["memory"],
+                },
+            }
+
+            result["results"].append(run_result)
+
+        return FakeJob(self, Result.from_dict(result))
+
+
+class MockRestlessFineAmp(MockRestlessBackend):
+    """A mock backend for restless single-qubit fine amplitude experiments."""
+
+    def __init__(
+        self, angle_error: float, angle_per_gate: float, gate_name: str, rng_seed: int = 0
+    ):
+        """Setup a mock backend to test the restless fine amplitude calibration.
+
+        Args:
+            angle_error: The rotation error per gate.
+            angle_per_gate: The angle per gate.
+            gate_name: The name of the gate to find in the circuit.
+            rng_seed: The random bit generator seed.
+        """
+        self.angle_error = angle_error
+        self._gate_name = gate_name
+        self._angle_per_gate = angle_per_gate
+        super().__init__(rng_seed=rng_seed)
+
+        self.configuration().basis_gates.extend(["sx", "x"])
+
+    def _compute_outcome_probabilities(self, circuits: List[QuantumCircuit]):
+        """Compute the probabilities of being in the excited state or
+        ground state for all circuits."""
+
+        self._precomputed_probabilities = {}
+
+        for idx, circuit in enumerate(circuits):
+
+            n_ops = circuit.count_ops().get(self._gate_name, 0)
+            angle = n_ops * (self._angle_per_gate + self.angle_error)
+
+            if self._gate_name != "sx":
+                angle += np.pi / 2 * circuit.count_ops().get("sx", 0)
+
+            if self._gate_name != "x":
+                angle += np.pi * circuit.count_ops().get("x", 0)
+
+            prob_0 = np.sin(angle / 2) ** 2
+            prob_1 = 1 - prob_0
```

Comment:
```suggestion
            prob_1 = np.sin(angle / 2) ** 2
            prob_0 = 1 - prob_1
```
These were flipped.

Reply:
Thanks, I modified this here: [55a8183](https://github.com/Qiskit/qiskit-experiments/pull/689/commits/55a818331f55628f71e9ea2cfcc709025d6734f4).